### PR TITLE
Add a timeout in RefreshContainers to avoid hanging connections

### PR DESCRIPTION
**Pull request by wsong**

### CODE REVIEW

**Comment by wsong** [comment]
Ping @jlhawn @alexmavr for review

File: cluster/engine.go [hunk 1]
```diff
@@ -728,7 +728,9 @@ func (e *Engine) RefreshContainers(full bool) error {
 		All:  true,
 		Size: false,
 	}
-	containers, err := e.apiClient.ContainerList(context.Background(), opts)
+	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
```
Comment (alexmavr) [inline cluster/engine.go]: The timeout value could be `requestTimeout` which is already a const in this file set to 30 seconds. That constant reflects the timeout for any request sent to an engine, so it's appropriate for this context

Comment (wsong) [reply cluster/engine.go]: Good call. Done.

**Comment by wsong** [comment]
Ping @nishanttotla

**Pull request merged**